server/publications/collections/shops.js

```diff
@@ -14,6 +15,11 @@ Meteor.publish("PrimaryShop", () => Shops.find({
 Meteor.publish("MerchantShops", function (shopsOfUser = Reaction.getShopsForUser(["admin"], this.userId)) {
   check(shopsOfUser, Array);
 
+  // Check if user has permissions for those shops
+  if (!isEqual(Reaction.getShopsForUser(["admin"], this.userId), shopsOfUser)) {
```

Comment:
Can we not use the `hasPermission` method here?

Reply:
@Akarshit

Reply:
@spencern AFAIK To use `hasPermission` here, I would have to do something like
```
const allow = shopsOfUser.reduce((allow, shop) => allow && Reaction.hasPermission(...), true);
```
The `getShopsForUser` kind of contains the above code, so I used it.

Though I can change it if you want.

Reply:
I think we should abstract this functionality into a common Reaction API then. We can use either method but should be well documented. Even with your comment on line 18, I'm not entirely clear on how that method is working which is why I thought using `hasPermissions` might work first.

Reply:
>  abstract this functionality into a common Reaction API

I didn't get what do you mean by this.

Reply:
@Akarshit I mean that if we have a common need to check the permissions of a user for all shops, we should probably have a Reaction method such as `Reaction.hasPermissionForAnyShop` or whatever the method is. That method could wrap the calls to getShops for users and be a well-documented way to accomplish that specific task.

Reply:
OK doing that.